evadb/functions/xgboost.py

```diff
@@ -0,0 +1,47 @@
+# coding=utf-8
+# Copyright 2018-2023 EvaDB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pickle
+
+import pandas as pd
+
+from evadb.functions.abstract.abstract_function import AbstractFunction
+from evadb.utils.generic_utils import try_to_import_xgboost
+
+
+class GenericXGBoostModel(AbstractFunction):
+    @property
+    def name(self) -> str:
+        return "GenericXGBoostModel"
+
+    def setup(self, model_path: str, **kwargs):
+        try_to_import_xgboost()
+
+        self.model = pickle.load(open(model_path, "rb"))
+
+    def forward(self, frames: pd.DataFrame) -> pd.DataFrame:
+        # Last column is the value to predict, hence don't pass that to the
+        # predict method.
+        predictions = self.model.predict(frames.iloc[:, :-1])
```

Comment:
There is no guarantee that last the column is the value to predict I think in this case. We need to store the column to predict in this case. You can again check the ludwig for example.

Reply:
But it seems that the auto_train methods for Ludwig and XGBoost are different. In Ludwig, you provide the entire dataset (X + Y) to the auto train method and then specify the column which is supposed to act as Y. Whereas XGBoost auto train specifies the feature matrix X and prediction variable Y.

Reply:
Could you elaborate what is the difference? 

What I meant originally is that the following query will not work, because the predict column is not the last one. 
```
CREATE FUNCTION IF NOT EXISTS PredictRent FROM
( SELECT number_of_rooms, number_of_bathrooms, days_on_market, rental_price FROM HomeRentals )
TYPE XGBoost
PREDICT 'number_of_rooms';
```

Reply:
As discussed, we now pass the prediction column to the .py model files